sky/backends/cloud_vm_ray_backend.py

```diff
@@ -4521,6 +4523,20 @@ def get_storage_mounts_metadata(
                 storage_metadata, sync_on_reconstruction=False)
         return storage_mounts
 
+    def _skypilot_predefined_envs(
+            self, handle: CloudVmRayResourceHandle) -> Dict[str, str]:
+        """Returns the SkyPilot predefined environment variables."""
+        return {
+            'SKYPILOT_CLUSTER_INFO': json.dumps({
+                'cluster_name': handle.cluster_name,
+                'num_nodes': handle.launched_nodes,
+                'cloud': str(handle.launched_resources.cloud),
+                'region': handle.launched_resources.region,
+                'zone': handle.launched_resources.zone,
```

Comment:
Shall we do a quick test on clouds that don't support zones (i.e., None here)?

Reply:
It works for azure by returning null for `echo $SKYPILOT_CLUSTER_INFO | jq .zone`.

Added several smoke tests. Please refer to the PR description.